apps/authentication-service/src/resources/clients/clientsRoutesHandlers.test.ts

```diff
@@ -0,0 +1,357 @@
+      // Makes sure we get a User without a Client
+      server.use(
+        http.get(AUTH0_USER_BY_ID, () =>
+          HttpResponse.json(exampleUserWithoutClient),
+        ),
+      );
+
+      const token = getTestToken();
+
+      const res = {
+        json: jest.fn(),
+        status: jest.fn().mockReturnThis(),
+      } as unknown as Response;
+
+      const req: Request = {
+        headers: {
+          authorization: `Bearer ${token}`,
+        },
+      } as Request;
+
+      await clientsGetV1(req, res);
+
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      expect(res.status).toHaveBeenCalledWith(404);
+      expect(res.json).toHaveBeenCalledWith({ message: "Client not found" });
+    });
+
+    it("should error when trying to get client, and the rate limit has been reached", async () => {
+      server.use(
+        http.get(
+          AUTH0_CLIENTS_BY_ID,
+          () => new HttpResponse(null, { status: 429 }),
+        ),
+      );
+
+      const token = getTestToken();
+
+      const res = {
+        json: jest.fn(),
+        status: jest.fn().mockReturnThis(),
+      } as unknown as Response;
+
+      const req: Request = {
+        headers: {
+          authorization: `Bearer ${token}`,
+        },
+      } as Request;
+
+      await clientsGetV1(req, res);
+
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      expect(res.status).toHaveBeenCalledWith(429);
+      expect(res.json).toHaveBeenCalledWith({ message: "Rate limit exceeded" });
+    });
+
+    it("should error when trying to get client, and and unknown error occurs", async () => {
+      server.use(
+        http.get(
+          AUTH0_CLIENTS_BY_ID,
+          () => new HttpResponse(null, { status: 500 }),
+        ),
+      );
+
+      const token = getTestToken();
+
+      const res = {
+        json: jest.fn(),
+        status: jest.fn().mockReturnThis(),
+      } as unknown as Response;
+
+      const req: Request = {
+        headers: {
+          authorization: `Bearer ${token}`,
+        },
+      } as Request;
+
+      await clientsGetV1(req, res);
+
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      expect(res.status).toHaveBeenCalledWith(500);
+      expect(res.json).toHaveBeenCalledWith({
+        message: "Unable to get client",
+      });
+    });
+  });
+
+  describe("clientsDeleteV1", () => {
+    it("should delete a client, based on the user token", async () => {
+      const token = getTestToken();
+
+      const res = {
+        send: jest.fn(),
+      } as unknown as Response;
+
+      const req: Request = {
+        headers: {
+          authorization: `Bearer ${token}`,
+        },
+      } as Request;
+
+      await clientsDeleteV1(req, res);
+
+      expect(res.send).toHaveBeenCalledWith("Client successfully deleted.");
+    });
+
+    it("should error when trying to delete a client, and the client id is invalid", async () => {
```

Comment:
Invalid makes it seems like there is a client, but the request failed. You mean missing right?